### DISCUSSION TITLE
Increase flexibility of "Require review from Code Owners" option

### DISCUSSION THREAD

**Discussion by BalestraPatrick** [community/community#44887] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

Hello!

We are heavy users of GitHub and the CODEOWNERS feature. As we scale our monorepo, we would like to give flexibility to our developers to require review from Code Owners to prevent changes to critical parts of our codebase. On the other hand, we don't want to require review everywhere because there are parts of the codebase that are owned by a single person or team for example, and requiring a review from that specific group would slow down big changes that span hundreds of components for example. There are solutions around this (such as having a catch-all group that can force approve on their behalf) but it's still not a great experience.

We would like to have more flexibility in which parts of the codebase require a code owner review. We would want the “Require review from Code Owners” option in the branch protection to be disabled, but an option to enable it per-path via (for example) the CODEOWNERS file. It could have the following format as an example:
```
very-critical-build-system-thing @infra-team #require-review-from-code-owners

regular/feature/that/can/be/reviewed/by/anyone/the/owner/just/needs/to/be/pinged @some-feature-team
```

For the first path review is assigned to and required by the @infra-team to allow merging. The #require-review-from-code-owners is just my attempt to visualize one way to enable that setting for that specific path.
While for the second path code review can be done by anyone with write (code review) permission.  @some-feature-team will just be pinged/assigned but their review isn’t blocking.

I hope my request is clear. This doesn't seem to be possible today but would really help us.

**Comment by queenofcorgis** [comment]
Hi there! Our team is tracking this internally, I've copied this feedback over to them! Thank you so much for sharing this with us and if anyone else has anything they'd like to be added please let me know. Cheers!
